docs/specifications/tests/Syntax-TP/syntax06.md

```diff
@@ -1,54 +1,131 @@
-in section 3.3 of [RFC 1034](https://tools.ietf.org/rfc/rfc1034.txt) and
-section 2.2 of [RFC 1912](https://tools.ietf.org/rfc/rfc1912.txt). The RNAME
-field should follow the rules of an e-mail address also defined in section
-3.4.1 of [RFC 2822](https://tools.ietf.org/html/rfc2822#section-3.4).
+in section 3.3.13 of [RFC 1035] and section 2.2 of [RFC 1912]. The RNAME
+field should follow the rules of an e-mail address also defined in 
+[RFC 5322, section 3.4.1].
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Ordered description of steps to be taken to execute the test case
+
+1. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
+
+2. Create a SOA query for the apex of the *Child Zone* with RD flag unset.
+
+3. For each name server in *Name Server IP* do:
+   1. Send the SOA query over UDP to the name server.
+   2. If the name server does not respond with a DNS response, then:
+      1. Emit *[NO_RESPONSE]*.
+      2. Go to next server.
+   3. If the DNS response does not include a SOA record in the 
+      answer section, then:
+      1. Emit *[NO_RESPONSE_SOA_QUERY]*.
+      2. Go to next server.
+   4. Extract the RNAME from the SOA record.
+   5. Convert the first "." without backslash quoting to an "@" in 
+      the RNAME.
+   6. Convert any backslash quoted "." to a single "." without quoting
+      (see section 5.1, 5.3 and 8 in [RFC1035] for the use of backslash).
+   7. If the converted string (mail address) does not meet the 
+      specification in [RFC 5322, section 3.4.1], then 
+      1. Emit *[RNAME_RFC822_INVALID]*.
+      2. Go to next server.
+   8. Extract the domain part (to the right of "@") from the mail 
+      address.
+   9. Create an MX query for the domain part and send it to a resolving
+       name server. 
+   10. If the MX lookup does not return a DNS response with with RCODE 
+       "NOERROR", then:
+       1. Emit *[RNAME_MAIL_DOMAIN_INVALID]*.
+       2. Go to next server.
+   11. If the MX lookup returned a CNAME or a chain of CNAMEs then
+       use the final name instead of the domain part in the steps below.
+   12. If the MX lookup returned a NO DATA response (no MX record), 
+       then 
+       1. Create address queries (A and AAAA) for the domain part and
+          send it to a resolving name server.
+       2. Disregard any A record with 127.0.0.1 or AAAA with ::1.
+       3. Disregard any A or AAAA records outside the Answer section.
+       4. If no A or AAAA was returned then emit 
+          *[RNAME_MAIL_DOMAIN_INVALID]*.
+   13. If the MX lookup returned one or more MX records, then for each
+       mail exchange (domain name in RDATA of the MX record) do:
+       1. Create address queries (A and AAAA) and send it to a 
+          resolving name server.
+       2. Disregard all A and AAAA records that do not have the same
+          owner name as the mail exchange name (i.e. do not follow
+          any CNAME).
```

Comment:
There's no corresponding step for this in the no-MX branch. Is that intentional?

Reply:
Yes.

When we ask for MX it is OK to follow CNAME and that is covered in "If the MX lookup returned a CNAME or a chain of CNAMEs then use the final name instead of the domain part in the steps below."

When we come to the non-MX we have already followed the CNAME and there cannot be any more CNAME to follow.

When we come to the MX then that MX (or those MX) are already based on the CNAME above (if any), and now we do a look-up of the mail exchange in the MX, and that must NOT follow CNAME.

Should I clarify?

Reply:
Ok, I see how you're thinking. I'm looking at this from a different angle. I'm looking at it from an algorithmic point of view as opposed to an educational point of view.

From an algorithmic point of view, these two cases behave differently. In the no-MX case you should consider A and AAAA records with any owner name, and in the MX case you should only consider A and AAAA records with a specific owner name. If the resolver responds with additional A and AAAA records with different owner names for whatever reason, the MX and no-MX cases are bound to behave differently.

I think it would be nice if both branches presented the same behavior. But I don't think it's critical.

Reply:
Will not update.